src/trip/TripContext.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { createContext, useState, useEffect } from 'react'
+
+import { getCredentials, register, unregister } from '../login/store'
+import { getCallApi, postCallApi, putCallApi, deleteCallApi } from '../utils/api/fetchMiddleware'
+import { USERS_ENDPOINT, TRIPS_ENDPOINT } from './constants'
+
+export const TripContext = createContext();
+
+const TripContextProvider = (props) => {
+    const [login, setLogin] = useState()
+    const [users, setUsers] = useState()
+    const [trips, setTrips] = useState()
+
+
+    const createTrip = async (trip) => { 
+        const response = await postCallApi(TRIPS_ENDPOINT, trip)
+        if (response.ok) {
+            const jsonData = await response.json()
+            setTrips([...trips, jsonData.data])
+        }
+    }
+
+    const updateTrip = async (id, trip) => {
+        const response = await putCallApi(`${TRIPS_ENDPOINT}/${id}`, trip)
+        if (response.ok) {
+            const jsonData = await response.json()
+            setTrips(prevTrips => prevTrips.map(trip => trip._id === jsonData.data._id ? jsonData.data : trip))
+        }
+    }
+
+    const deleteTrip = async (id) => {
+        const response = await deleteCallApi(`${TRIPS_ENDPOINT}/${id}`)
+        if (response.ok) {
+            const jsonData = await response.json()
+            setTrips(prevTrips => prevTrips.filter(trip => trip._id !== jsonData.data._id))
```

Comment:
is it normal/intended that you pass a function to setTrips instead of values ?

Reply:
Yes, it is intended, it is convenient when new state is calculated based on the previous state, see in the doc below:

![maj_func_setState](https://user-images.githubusercontent.com/10847455/90044205-9b4ad300-dccd-11ea-918e-88eb2455f589.png)

Reply:
Didn't know that, cool !